backend/src/controllers/groups-controller.ts

```diff
@@ -77,6 +84,21 @@ export const updateName = async (req: Request, res: Response) =>{
     if(existingGroup){
         return res.status(409).json({ error: "This group name is already taken"});
     }
+    //Checks if UserId exists
+    if(!req.token?.userId){
+        return res.status(422).json({error: "UserId does not exist"});
+    }
+
+    const groupOwner = req.body.userId;
+    const findGroupOwnerById = Group.find({ userId: groupOwner });
```

Comment:
This will return a list of groups that the user owns instead of checking that the user owns `groupId`.

`.findOne({userId: req.token.userId, _id: groupID})` should fix this.